test/callback.jl

```diff
@@ -228,9 +228,11 @@ context("With solver $(typeof(infosolver))") do
     # Test that solver fills solution correctly
     function myinfo(cb)
         entered[1] = true
-        push!(nodes,      MathProgBase.cbgetexplorednodes(cb))
-        push!(objs,       MathProgBase.cbgetobj(cb))
-        push!(bestbounds, MathProgBase.cbgetbestbound(cb))
+        if MathProgBase.cbgetstate(cb) == :Other
+            push!(nodes,      MathProgBase.cbgetexplorednodes(cb))
```

Comment:
Previously, the callback will only fire when the state is `:MIPInfo` (which is supposed to be `:Other` in MPB-land). It's to let current tests pass, since Gurobi.jl does not allow for `MPB.cbgetobj` to work when the state is `:MIPSol`.